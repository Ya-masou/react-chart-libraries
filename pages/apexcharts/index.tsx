// import Chart from "react-apexcharts";
import { useMemo } from "react";
import type { ApexOptions } from "apexcharts";
import dynamic from "next/dynamic";

// https://nextjs.org/docs/advanced-features/dynamic-import
const Chart = dynamic(() => import("react-apexcharts"), { ssr: false });

export function BarChart() {
  const options: ApexOptions = {
    chart: {
      id: "basic-bar",
      toolbar: {
        show: false,
      },
      animations: {
        enabled: false,
      },
      height: "800px",
    },
    xaxis: {
      categories: [1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999],
    },
    dataLabels: {
      enabled: false,
    },
  };

  const series: ApexOptions["series"] = [
    {
      name: "series-1",
      data: [30, 40, 45, 50, 49, 60, 70, 91, 8],
      type: "line",
      color: "rgba(255, 0, 0, 0.2)",
    },
    {
      name: "series-2",
      data: [20, 10, 40, 19, 30, 30, 20, 100, 80],
      type: "bar",
    },
  ];

  return <Chart series={series} options={options} />;
}

function LineChart() {
  const options: ApexOptions = {
    chart: {
      id: "basic-line",
      toolbar: {
        show: false,
      },
      animations: {
        easing: "linear",
        speed: 200,
        animateGradually: {
          enabled: false,
        },
        dynamicAnimation: {
          enabled: true,
          speed: 200,
        },
      },
      zoom: {
        enabled: false,
      },
    },
    yaxis: {
      showAlways: true,
      max: (max) => max + 10,
    },
    xaxis: {
      categories: [1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999],
    },
    dataLabels: {
      enabled: false,
    },
  };

  const series: ApexOptions["series"] = [
    {
      // name: "series-1",
      name: "series-1",
      data: [30, 40, 45, 50, 49, 60, 70, 91, 8],
      type: "line",
    },
    {
      name: "series-2",
      data: [20, 10, 40, 19, 30, 30, 20, 100, 80],
    },
  ];

  return <Chart series={series} options={options} type="line" />;
}

function RadarChart() {
  const series: ApexOptions["series"] = [
    {
      name: "Series 1",
      data: [80, 50, 30, 40, 100, 20],
    },
    {
      name: "Series 2",
      data: [20, 30, 40, 80, 20, 80],
    },
    {
      name: "Series 3",
      data: [44, 76, 78, 13, 43, 10],
    },
  ];

  const options: ApexOptions = {
    chart: {
      id: "radar",
      dropShadow: {
        enabled: true,
        blur: 1,
        left: 1,
        top: 1,
      },
    },
    title: {
      text: "Radar Chart - Multi Series",
    },
    stroke: {
      width: 2,
    },
    fill: {
      opacity: 0.1,
    },
    markers: {
      size: 0,
    },
    xaxis: {
      categories: ["2011", "2012", "2013", "2014", "2015", "2016"],
    },
    yaxis: {
      min: 0,
      showAlways: true,
    },
  };

  return <Chart series={series} options={options} type="radar" />;
}

function PieChart() {
  const options: ApexOptions = {
    chart: {
      id: "basic-pie",
      toolbar: {
        show: false,
      },
      zoom: {
        enabled: false,
      },
    },
    title: {
      text: "ϵ( 'Θ' )϶",
      align: "center",
    },
    // xaxis: {
    //   categories: [1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999],
    // },
    // dataLabels: {
    //   enabled: false,
    // },
    labels: ["Team A", "Team B", "Team C", "Team D", "Team E"],
    colors: ["black", "cyan", "magenta", "yellow", "blue"],
    dataLabels: {
      style: {
        colors: ["black", "cyan", "magenta", "yellow", "blue"],
      },
    },
    stroke: {
      colors: ["black", "cyan", "magenta", "yellow", "blue"],
    },
    fill: {
      colors: [
        "#F44336",
        "#E91E63",
        "green",
        "hsl(210, 100%, 80%)",
        "rgba(255, 0, 0, 0.1)",
      ],
    },
  };

  const series: ApexOptions["series"] = [44, 55, 13, 43, 22];

  return <Chart series={series} options={options} type="pie" />;
}

export default function Apex() {
  return (
    <main
      style={{
        padding: "1rem",
      }}
    >
      <BarChart />
      <hr />
      <LineChart />
      <hr />
      <RadarChart />
      <hr />
      <PieChart />
    </main>
  );
}
