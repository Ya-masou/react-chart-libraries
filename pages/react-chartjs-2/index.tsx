import { Bar, Line, Radar, Pie } from "react-chartjs-2";
import { ChartData, ChartOptions } from "chart.js";

function BarChart() {
  const data: ChartData<"bar"> = {
    labels: ["Red", "Blue", "Yellow", "Green", "Purple", "Orange"],
    datasets: [
      {
        label: "# of Votes",
        data: [12, 19, 3, 5, 2, 3],
        backgroundColor: [
          "rgba(255, 99, 132, 0.2)",
          "rgba(54, 162, 235, 0.2)",
          "rgba(255, 206, 86, 0.2)",
          "rgba(75, 192, 192, 0.2)",
          "rgba(153, 102, 255, 0.2)",
          "rgba(255, 159, 64, 0.2)",
        ],
        borderColor: [
          "rgba(255, 99, 132, 1)",
          "rgba(54, 162, 235, 1)",
          "rgba(255, 206, 86, 1)",
          "rgba(75, 192, 192, 1)",
          "rgba(153, 102, 255, 1)",
          "rgba(255, 159, 64, 1)",
        ],
        borderWidth: 1,
      },
    ],
  };

  const options: ChartOptions<"bar"> = {
    scales: {
      y: {
        beginAtZero: true,
      },
    },
  };

  return (
    <Bar
      data={data}
      options={options}
      // getDatasetAtEvent={(dataset, event) => {
      //   console.log(dataset);
      //   console.log(event);
      // }}
      getElementsAtEvent={(element, event) => {
        console.log(element);
        console.log(event);
      }}
    />
  );
}

function LineChart() {
  const data: ChartData<"line"> = {
    labels: ["1月", "2月", "3月", "4月", "5月", "6月"],
    datasets: [
      {
        label: "東京",
        data: [12, 19, 3, 5, 2, 3],
        borderWidth: 1,
      },
      {
        label: "神奈川",
        data: [4, 15, 21, 5, 10, 1],
        borderWidth: 1,
        borderColor: "green",
      },
      {
        label: "群馬",
        data: [200, 100, 30, 100, 120, 80],
        backgroundColor: "black",
        borderWidth: 2,
        borderColor: "red",
        pointBorderColor: "red",
        pointBackgroundColor: "red",
        pointHoverBackgroundColor: "cyan",
      },
    ],
  };

  const options: ChartOptions<"line"> = {
    showLine: true,
    scales: {
      x: {
        // title: {
        //   display: boolean;
        //   text: string | string[];
        //   color: Color;
        //   font: FontSpec;
        //   padding: number | {
        //     top: number;
        //     bottom: number;
        //   };
        // };
        ticks: {
          color: "green",
          font: {
            weight: "bold",
          },
        },
      },
      y: {
        title: {
          display: true,
          text: "あいー",
          font: {
            size: 30,
          },
        },
        ticks: {
          font: {
            size: 30,
          },
          callback: function (value, index, values) {
            return value + "人";
          },
        },
      },
    },
    plugins: {
      title: {
        text: "感染者数",
        display: true,
        font: {
          size: 20,
        },
      },
      legend: {
        labels: {
          font: {
            size: 24,
          },
        },
      },
    },
  };

  return <Line data={data} options={options} />;
}

function RadarChart() {
  const data: ChartData<"radar"> = {
    labels: [
      "やる気",
      "だるさ",
      "腰が重い",
      "重いものを持てる",
      "免許持ってる",
      "遠い",
    ],
    datasets: [
      {
        label: "東京",
        data: [12, 19, 3, 5, 2, 3],
        borderWidth: 1,
        borderColor: "rgba(0, 0, 255, 0.5)",
        backgroundColor: "rgba(0, 0, 255, 0.1)",
      },
      {
        label: "神奈川",
        data: [4, 15, 21, 5, 10, 1],
        borderWidth: 1,
        borderColor: "rgba(0, 255, 0, 0.5)",
        backgroundColor: "rgba(0, 255, 0, 0.1)",
      },
      {
        label: "群馬",
        data: [40, 20, 10, 21, 8, 2],
        borderWidth: 1,
        borderColor: "rgba(255, 0, 0, 0.5)",
        backgroundColor: "rgba(255, 0, 0, 0.1)",
      },
    ],
  };

  const options: ChartOptions<"radar"> = {
    plugins: {
      legend: {
        position: "bottom",
        labels: {
          // This more specific font property overrides the global property
          font: {
            size: 14,
          },
        },
      },
    },
    scales: {
      r: {
        ticks: {
          font: {
            size: 16,
          },
        },
        pointLabels: {
          color: "red",
          font: {
            size: 24,
            weight: "bold",
          },
        },
      },
    },
  };

  return <Radar data={data} options={options} />;
}

function PieChart() {
  const data: ChartData<"pie"> = {
    labels: [
      "やる気",
      "だるさ",
      "腰が重い",
      "重いものを持てる",
      "免許持ってる",
      "遠い",
    ],
    datasets: [
      {
        label: "群馬",
        data: [40, 20, 10, 21, 8, 2],
        borderWidth: 0,
        backgroundColor: ["red", "yellow", "blue", "green", "orange", "pink"],
      },
    ],
  };

  return <Pie data={data} />;
}

export default function ReactChartjs2Page() {
  return (
    <>
      <main
        style={{
          padding: "1rem",
        }}
      >
        <BarChart />
        <hr />
        <LineChart />
        <hr />
        <RadarChart />
        <hr />
        <PieChart />
      </main>
    </>
  );
}
