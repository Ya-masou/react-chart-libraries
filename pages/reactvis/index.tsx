import "react-vis/dist/style.css";
import {
  XYPlot,
  VerticalGridLines,
  HorizontalGridLines,
  XAxis,
  YAxis,
  VerticalBarSeries,
  LabelSeries,
  FlexibleXYPlot,
  LineSeries,
} from "react-vis";

function BarChart() {
  const data = [
    { x: "A", y: 10 },
    { x: "B", y: 5 },
    { x: "C", y: 1500 },
  ];

  const labelData = data.map((d, idx) => ({
    x: d.x,
    y: d.y,
  }));

  return (
    <FlexibleXYPlot xType="ordinal" Distance={100}>
      <VerticalGridLines />
      <HorizontalGridLines />
      <XAxis />
      <YAxis />
      <VerticalBarSeries data={data} />
      <LabelSeries data={labelData} getLabel={(d) => d.x} />
    </FlexibleXYPlot>
  );
}

function LineChart() {
  const data = [
    { x: "A", y: 10 },
    { x: "B", y: 5 },
    { x: "C", y: 1500 },
  ];

  const labelData = data.map((d, idx) => ({
    x: d.x,
    y: d.y,
  }));

  return (
    <FlexibleXYPlot xType="ordinal" Distance={100}>
      <VerticalGridLines />
      <HorizontalGridLines />
      <XAxis />
      <YAxis />
      <LineSeries data={data} />
      <LabelSeries data={labelData} getLabel={(d) => d.x} />
    </FlexibleXYPlot>
  );
}

function RadarChart() {
  return <p>RadarChart</p>;
}

function PieChart() {
  return <p>PieChart</p>;
}

export default function Reactvis() {
  return (
    <main
      style={{
        padding: "1rem",
      }}
    >
      <BarChart />
      <hr />
      <LineChart />
      <hr />
      <RadarChart />
      <hr />
      <PieChart />
    </main>
  );
}
