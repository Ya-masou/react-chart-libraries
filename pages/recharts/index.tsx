import {
  BarChart,
  Bar,
  Cell,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
  ResponsiveContainer,
  LineChart,
  Line,
  RadarChart,
  Radar,
  PolarGrid,
  PolarAngleAxis,
  PolarRadiusAxis,
  PieChart,
  Pie,
} from "recharts";

function MyBarChart() {
  const data = [
    {
      name: "Page A",
      uv: 4000,
      pv: 2400,
      amt: 2400,
    },
    {
      name: "Page B",
      uv: 3000,
      pv: 1398,
      amt: 2210,
    },
    {
      name: "Page C",
      uv: 2000,
      pv: 9800,
      amt: 2290,
    },
    {
      name: "Page D",
      uv: 2780,
      pv: 3908,
      amt: 2000,
    },
    {
      name: "Page E",
      uv: 1890,
      pv: 4800,
      amt: 2181,
    },
    {
      name: "Page F",
      uv: 2390,
      pv: 3800,
      amt: 2500,
    },
    {
      name: "Page G",
      uv: 3490,
      pv: 4300,
      amt: 2100,
    },
  ];

  return (
    <div
      style={{
        width: "100%",
        height: 300,
        // aspectRatio: "16 / 9",
      }}
    >
      <ResponsiveContainer width="100%" height="100%">
        <BarChart
          // width={500}
          // height={300}
          data={data}
          margin={{
            top: 0,
            right: 0,
            left: 0,
            bottom: 0,
          }}
        >
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip />
          <Legend />
          <Bar dataKey="pv" stackId="a" fill="#8884d8" />
          <Bar dataKey="uv" stackId="a" fill="#82ca9d" />
        </BarChart>
      </ResponsiveContainer>
    </div>
  );
}

function MyLineChart() {
  const data = [
    {
      name: "Page A",
      uv: 4000,
      pv: 2400,
      amt: 2400,
    },
    {
      name: "Page B",
      uv: 3000,
      pv: 1398,
      amt: 2210,
    },
    {
      name: "Page C",
      uv: 2000,
      pv: 9800,
      amt: 2290,
    },
    {
      name: "Page D",
      uv: 2780,
      pv: 3908,
      amt: 2000,
    },
    {
      name: "Page E",
      uv: 1890,
      pv: 4800,
      amt: 2181,
    },
    {
      name: "Page F",
      uv: 2390,
      pv: 3800,
      amt: 2500,
    },
    {
      name: "Page G",
      uv: 3490,
      pv: 4300,
      amt: 2100,
    },
  ];
  return (
    <div
      style={{
        width: "100%",
        height: 400,
      }}
    >
      <ResponsiveContainer width="100%" height="100%">
        <LineChart
          // width={500}
          // height={300}
          data={data}
          margin={{
            top: 5,
            right: 30,
            left: 20,
            bottom: 5,
          }}
        >
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="name" />
          <YAxis />
          <Tooltip
          // content={(props) => {
          //   if (props.active) {
          //     console.log(props);
          //   }
          //   return <p>ϵ( 'Θ' )϶</p>;
          // }}
          />
          <Legend onClick={(event) => console.log(event)} />
          <Line
            type="monotone"
            dataKey="pv"
            stroke="#8884d8"
            activeDot={false}
            onMouseOver={(event) => console.log(event)}
            strokeWidth={5}
          />
          <Line
            type="monotone"
            dataKey="uv"
            stroke="#82ca9d"
            onMouseOver={(event) => console.log(event)}
          />
        </LineChart>
      </ResponsiveContainer>
    </div>
  );
}

function MyRadarChart() {
  const data = [
    {
      subject: "Math",
      A: 120,
      B: 110,
      fullMark: 150,
    },
    {
      subject: "Chinese",
      A: 98,
      B: 130,
      fullMark: 150,
    },
    {
      subject: "English",
      A: 86,
      B: 130,
      fullMark: 150,
    },
    {
      subject: "Geography",
      A: 99,
      B: 100,
      fullMark: 150,
    },
    {
      subject: "Physics",
      A: 85,
      B: 90,
      fullMark: 150,
    },
    {
      subject: "History",
      A: 65,
      B: 85,
      fullMark: 150,
    },
  ];

  return (
    <div
      style={{
        width: "100%",
        height: 400,
      }}
    >
      <ResponsiveContainer width="100%" height="100%">
        <RadarChart cx="50%" cy="50%" outerRadius="80%" data={data}>
          <PolarGrid />
          <PolarAngleAxis dataKey="subject" />
          <PolarRadiusAxis angle={90} domain={[0, 150]} />
          <Radar
            name="Mike"
            dataKey="A"
            stroke="#8884d8"
            fill="#8884d8"
            fillOpacity={0.3}
            strokeWidth={4}
          />
          <Radar
            name="Lily"
            dataKey="B"
            stroke="#82ca9d"
            fill="#82ca9d"
            fillOpacity={0.3}
          />
          <Legend />
        </RadarChart>
      </ResponsiveContainer>
    </div>
  );
}

function MyPieChart() {
  const data = [
    { name: "Group A", value: 400 },
    { name: "Group B", value: 300 },
    { name: "Group C", value: 300 },
    { name: "Group D", value: 200 },
  ];
  const COLORS = ["#0088FE", "#00C49F", "#FFBB28", "#FF8042"];

  return (
    <div style={{ width: "100%", height: 300 }}>
      <ResponsiveContainer>
        <PieChart>
          <Pie dataKey="value" data={data} label>
            {data.map((_entry, index) => (
              <Cell
                key={`cell-${index}`}
                fill={COLORS[index % COLORS.length]}
              />
            ))}
          </Pie>
        </PieChart>
      </ResponsiveContainer>
    </div>
  );
}

export default function Recharts() {
  return (
    <main
      style={{
        padding: "1rem",
      }}
    >
      <MyBarChart />
      <hr />
      <MyLineChart />
      <hr />
      <MyRadarChart />
      <hr />
      <MyPieChart />
    </main>
  );
}
