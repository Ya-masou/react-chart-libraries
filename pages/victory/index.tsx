import {
  VictoryChart,
  VictoryBar,
  VictoryAxis,
  VictoryTheme,
  VictoryStack,
  VictoryTooltip,
  VictoryLine,
  VictoryScatter,
  VictoryArea,
  VictoryPolarAxis,
  VictoryPie,
} from "victory";

export function BarChart() {
  const data1 = [
    { quarter: 1, earnings: 13000, label: "data1 の quarter1" },
    { quarter: 2, earnings: 16500, label: "data1 の quarter2" },
    { quarter: 3, earnings: 14250, label: "data1 の quarter3" },
    { quarter: 4, earnings: 19000, label: "data1 の quarter4" },
  ];
  const data2 = [
    { quarter: 1, earnings: 13000, label: "data2 の ✊" },
    { quarter: 2, earnings: 16500, label: "data2 の quarter2" },
    { quarter: 3, earnings: 14250, label: "data2 の quarter3" },
    { quarter: 4, earnings: 19000, label: "data2 の quarter4" },
  ];

  return (
    <VictoryChart domainPadding={20} theme={VictoryTheme.material}>
      <VictoryAxis
        // tickValues specifies both the number of ticks and where
        // they are placed on the axis
        tickValues={[1, 2, 3, 4]}
        tickFormat={["Quarter 1", "Quarter 2", "Quarter 3", "Quarter 4"]}
      />
      <VictoryAxis
        dependentAxis
        // tickFormat specifies how ticks should be displayed
        tickFormat={(x) => `$${x / 1000}k`}
      />
      <VictoryStack colorScale="cool">
        <VictoryBar
          data={data1}
          // data accessor for x values
          x="quarter"
          // data accessor for y values
          y="earnings"
          labelComponent={<VictoryTooltip cornerRadius={8} />}
          style={{
            data: {
              fill: "tomato",
              strokeWidth: "4px",
              stroke: "red",
              width: 20,
            },
          }}
        />
        <VictoryBar
          data={data2}
          // data accessor for x values
          x="quarter"
          // data accessor for y values
          y="earnings"
        />
      </VictoryStack>
    </VictoryChart>
  );
}

export function LineChart() {
  interface Data {
    x: number;
    y: number;
    label: string;
  }
  const data = [
    { x: 1, y: 13000, label: "1" },
    { x: 2, y: 16500, label: "2" },
    { x: 3, y: 14250, label: "3" },
    { x: 4, y: 19000, label: "4" },
  ];
  return (
    <VictoryChart theme={VictoryTheme.material}>
      <VictoryLine
        style={{
          data: { stroke: "#c43a31" },
          parent: { border: "1px solid #ccc" },
        }}
        data={data}
        labels={({ datum }) => datum.x}
      />
      <VictoryScatter
        data={data}
        style={{
          data: {
            fill: "#c43a31",
          },
        }}
        labels={({ data }: { data: Data }) =>
          `${data.label} / ${data.x} & ${data.y}`
        }
        labelComponent={<VictoryTooltip />}
      />
    </VictoryChart>
  );
}

export function RadarChart() {
  const data = [
    { x: 1, y: 13000, label: "1" },
    { x: 2, y: 16500, label: "2" },
    { x: 3, y: 14250, label: "3" },
    { x: 4, y: 19000, label: "4" },
  ];

  return (
    <VictoryChart polar theme={VictoryTheme.material}>
      <VictoryArea data={data} />
      <VictoryPolarAxis />
    </VictoryChart>
  );
}

export function PieChart() {
  return (
    <VictoryPie
      data={[
        { x: "Cats", y: 35 },
        { x: "Dogs", y: 40 },
        { x: "Birds", y: 55 },
      ]}
      colorScale={["red", "orange", "green"]}
      // labelComponent={<VictoryTooltip />}
    />
  );
}

export default function Victory() {
  return (
    <main
      style={{
        padding: "1rem",
      }}
    >
      <BarChart />
      <hr />
      <LineChart />
      <hr />
      <RadarChart />
      <hr />
      <PieChart />
    </main>
  );
}
