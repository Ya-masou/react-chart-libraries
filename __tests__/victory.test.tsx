import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { PieChart } from "../pages/victory";
// import cy from 'cypress';

describe("Home", () => {
  it("renders a PieChart", () => {
    const { container } = render(<PieChart />);

    console.log(container.outerHTML);

    expect(container.querySelector("svg")).toBeInTheDocument();
  });

  it("renders PieChart unchanged", () => {
    const tree = renderer.create(<PieChart />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
