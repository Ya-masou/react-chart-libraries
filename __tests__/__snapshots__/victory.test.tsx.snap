// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders PieChart unchanged 1`] = `
<div
  className="VictoryContainer"
  style={
    Object {
      "height": "100%",
      "pointerEvents": "none",
      "position": "relative",
      "touchAction": "none",
      "width": "100%",
    }
  }
>
  <svg
    height={400}
    role="img"
    style={
      Object {
        "height": "100%",
        "pointerEvents": "all",
        "width": "100%",
      }
    }
    viewBox="0 0 400 400"
    width={400}
  >
    <g>
      <path
        d="M9.184850993605149e-15,-150A150,150,0,0,1,148.9063311147081,18.08050203829844L0,0Z"
        role="presentation"
        shapeRendering="auto"
        style={
          Object {
            "fill": "red",
            "padding": 10,
            "stroke": "transparent",
            "strokeWidth": 1,
          }
        }
        transform="translate(200, 200)"
      />
      <path
        d="M148.9063311147081,18.08050203829844A150,150,0,0,1,-69.70847580656522,132.81840384798153L0,0Z"
        role="presentation"
        shapeRendering="auto"
        style={
          Object {
            "fill": "orange",
            "padding": 10,
            "stroke": "transparent",
            "strokeWidth": 1,
          }
        }
        transform="translate(200, 200)"
      />
      <path
        d="M-69.70847580656522,132.81840384798153A150,150,0,0,1,-2.7554552980815446e-14,-150L0,0Z"
        role="presentation"
        shapeRendering="auto"
        style={
          Object {
            "fill": "green",
            "padding": 10,
            "stroke": "transparent",
            "strokeWidth": 1,
          }
        }
        transform="translate(200, 200)"
      />
      <text
        direction="inherit"
        dx={0}
        id="pie-labels-0"
        x={327}
        y={91.97}
      >
        <tspan
          dx={0}
          dy={0}
          style={
            Object {
              "fill": "#252525",
              "fontFamily": "'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif",
              "fontSize": 14,
              "letterSpacing": "normal",
              "padding": 20,
              "stroke": "transparent",
            }
          }
          textAnchor="start"
          x={327}
        >
          Cats
        </tspan>
      </text>
      <text
        direction="inherit"
        dx={0}
        id="pie-labels-1"
        x={279}
        y={362.97}
      >
        <tspan
          dx={0}
          dy={0}
          style={
            Object {
              "fill": "#252525",
              "fontFamily": "'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif",
              "fontSize": 14,
              "letterSpacing": "normal",
              "padding": 20,
              "stroke": "transparent",
            }
          }
          textAnchor="middle"
          x={279}
        >
          Dogs
        </tspan>
      </text>
      <text
        direction="inherit"
        dx={0}
        id="pie-labels-2"
        x={35}
        y={163.97}
      >
        <tspan
          dx={0}
          dy={0}
          style={
            Object {
              "fill": "#252525",
              "fontFamily": "'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif",
              "fontSize": 14,
              "letterSpacing": "normal",
              "padding": 20,
              "stroke": "transparent",
            }
          }
          textAnchor="end"
          x={35}
        >
          Birds
        </tspan>
      </text>
    </g>
  </svg>
  <div
    style={
      Object {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 99,
      }
    }
  >
    <svg
      height={400}
      style={
        Object {
          "height": "100%",
          "overflow": "visible",
          "width": "100%",
        }
      }
      viewBox="0 0 400 400"
      width={400}
    />
  </div>
</div>
`;
